### PR TITLE
Make parsed tree more consistent

### PR DESCRIPTION
**Pull request by aew**
- Modify grammar to simplify parser with ~30% fewer "special" functions and renamings
- Move to a consistent parse tree tracking the type of the node.  Although more human readable and easier to interact with for lookups, using maps of key-value pairs for the parsed tree that comes out of instaparse's transform function for things like type-fields, variables, and arguments actually makes it harder to do validation (e.g., guard against duplicate keys) and is less consistent to walk the tree (things like zippers tracking the path into the tree may be useful for clojure.spec global naming, and could simplify type lookup and validation).  Use a vector of maps instead.
  - Statements
    - Before: https://gist.github.com/aew/9a8508da64c03a8f26157de27ead9534
    - After: https://gist.github.com/aew/1bfa657743caa025e28d4fe83516c81d
  - Schemas
    - Before: https://gist.github.com/aew/d7ebe142283d37585761fdd859445f11
    - After: https://gist.github.com/aew/362da62a1cb949436a62f865118f83c6
- Fix root cause of the enum value ambiguity with boolean values.  The spec states: `EnumValue
  can be a Name but not true or false or null`
- Consolidate parser tests (including CATS cases and others) in a single test namespace that can react to changes in the parser namespace (for things like prism)
- Fix `:ListType` parsing
- Use `:argument-name` and `:field-name` to remain unambiguous.  Arguments can have both a `:variable-name` and an `:argument-name` and these were colliding when they were both called `:name`
- Simplify `:selection-set` parsed representation by removing the `:selection` and `:field` wrapping keys
  - Before: https://gist.github.com/aew/9a8508da64c03a8f26157de27ead9534#file-gistfile1-txt-L7
  - After: https://gist.github.com/aew/1bfa657743caa025e28d4fe83516c81d#file-gistfile1-txt-L10
- Simplify enum values in arguments by removing the `:enum-value` wrapper
  - Before: https://gist.github.com/aew/9a8508da64c03a8f26157de27ead9534#file-gistfile1-txt-L521
  - After: https://gist.github.com/aew/1bfa657743caa025e28d4fe83516c81d#file-gistfile1-txt-L489

### CODE REVIEW

File: test/graphql_clj/introspection_test.clj [hunk 1]
```diff
@@ -37,4 +35,4 @@ schema {
         context nil
         query "query { __schema { types {name kind} }}"
         result (executor/execute context schema resolver-fn query)]
-    (is (not (nil? (:data result))))))
+    (is (= {"__schema" nil} (:data result)))))              ;; TODO is this the expected result?
```
Comment (aew) [inline test/graphql_clj/introspection_test.clj]: I think this is a regression caused by this pull request, but I'm not sure I understand the output of this test on master either.  To investigate before merging.

**Comment by aew** [comment]
Also - this looks like another bug: https://gist.github.com/aew/362da62a1cb949436a62f865118f83c6#file-gistfile1-txt-L266 as the value is supposed to be an enum value of `JEDI` in this case.  To investigate before merging.

**Comment by aew** [comment]
This is ok to merge from my side at this point, as introspection is working (with a new test).

**Comment by tendant** [comment]
This is great. Thanks a lot!

**Pull request merged**